internal/build_defs.bzl

```diff
@@ -212,10 +218,10 @@ ts_library = rule(
         ),
         "tsickle_typed": attr.bool(default = True),
         "internal_testing_type_check_dependencies": attr.bool(default = False, doc = "Testing only, whether to type check inputs that aren't srcs."),
-        # @// is special syntax for the "main" repository
-        # The default assumes the user specified a target "node_modules" in their
-        # root BUILD file.
-        "node_modules": attr.label(default = Label("@//:node_modules")),
+        "node_modules": attr.label(
+            doc = """TODO(gregmagolan): doc string""",
+            default = Label("@npm//:typescript__dts"),
```

Comment:
This seems like the best way to avoid user's having to add `@npm//:typescript` as a dep to all `ts_library` targets since the default libs from the typescript package are always (almost always?) needed by ts_library. If a user isn't using bazel managed deps and doesn't have an `@npm` workspace then he'll have to override both the `compiler` and the `node_modules` attributes of ts_library and ng_module.

Here's the change needed in rules_nodejs to support this: https://github.com/bazelbuild/rules_nodejs/pull/328. We include only the dts files because we don't want all of the typescript .ts, .js & .json files being included as action inputs.

Reply:
I wonder if it would also make sense to include `node_modules/@types/**/*.d.ts` in this default node_modules filegroup since then all types would be available as would be default in typescript outside of bazel?

Reply:
An `@npm//:@types` scoped target is now generated that includes all of the types. I think its better to be explicit and not include these by default. User's that don't have `types = []` in their tsconfig can just depend on `@npm//:@types` to get all types into the compile.